### PR TITLE
device survey data

### PR DESCRIPTION
**Pull request by kaeline**
https://github.com/Metatavu/oss-surveys-api-spec/issues/21

### CODE REVIEW

File: swagger.yaml [hunk 1]
```diff
@@ -657,7 +657,7 @@ paths:
     get:
       operationId: findDeviceSurvey
       summary: Find a Device Survey.
-      description: Finds a Device Survey by id. Devices access this endpoint.
+      description: Finds a Device Survey by id.
       tags:
         - DeviceSurveys
       security:
```
Comment (villejuutila) [inline swagger.yaml]: Remove this endpoint specific authentication definition now that devices don't access this endpoint anymore.

**Review by villejuutila — APPROVED** [review]

File: swagger.yaml (hunk 1 above)
Comment (kaeline) [reply swagger.yaml]: Done

File: swagger.yaml [hunk 2]
```diff
@@ -1233,4 +1309,59 @@ components:
...
         - UPDATE
-        - DELETE
\ No newline at end of file
+        - DELETE
+
+    DeviceSurveyData:
+      type: object
+      description: Survey object that is sent to devices and contains only necessary info for display
+      properties:
+        id:
```
Comment (anttileppa) [inline swagger.yaml:1318]: id, pages, surveyId, deviceId and status could be marked as required and all should be marked as readOnly (this is readOnly entity)

File: swagger.yaml [hunk 3]
```diff
@@ -1233,4 +1309,59 @@ components:
...
+          type: string
+          format: uuid
+          description: ID of survey page
+          readOnly: true
+        layoutHtml:
+          type: string
+          description: Read-only version of page layout's HTML
+          readOnly: true
+        properties:
```
Comment (anttileppa) [inline swagger.yaml:1379]: could be readOnly?

File: swagger.yaml [hunk 4]
```diff
@@ -1233,4 +1309,59 @@ components:
...
+          format: date-time
+          description: Publication end time
+        metadata:
+          $ref: "#/components/schemas/Metadata"
+
+    DeviceSurveyPageData:
+      type: object
+      description: Survey page object that is sent to the device
+      properties:
+        id:
```
Comment (anttileppa) [inline swagger.yaml:1366]: id, layoutHtml and properties could be required (this is a readOnly entity)

**Review by anttileppa — APPROVED** [review]

**Pull request merged**